packages/cli/generation/remote-generation/remote-workspace-runner/src/SourceUploader.ts

```diff
@@ -0,0 +1,141 @@
+import { AbsoluteFilePath, join, RelativeFilePath } from "@fern-api/fs-utils";
+import { ApiDefinitionSource, SourceConfig } from "@fern-api/ir-sdk";
+import { loggingExeca } from "@fern-api/logging-execa";
+import { InteractiveTaskContext } from "@fern-api/task-context";
+import { FernWorkspace, IdentifiableSource } from "@fern-api/workspace-loader";
+import { FernRegistry as FdrAPI } from "@fern-fern/fdr-cjs-sdk";
+import { readFile, unlink } from "fs/promises";
+import tmp from "tmp-promise";
+
+const PROTOBUF_ZIP_FILENAME = "proto.zip";
+
+export type SourceType = "asyncapi" | "openapi" | "protobuf";
```

Comment:
does this work for a fern definition?

Reply:
No - the original `Source` is only present whenever we map from some OSS source into a Fern definition ([link](https://github.com/fern-api/fern/blob/d0a717de8a4776ddb949d93e8c31f603eca5ace0/packages/cli/workspace-loader/src/types/Workspace.ts#L25)). Otherwise we just assume the Fern definition file is the source.

The idea here was that we would only be uploading OpenAPI and Protobuf sources alongside the Fern definition files already included in the same register call.

If we want to separately upload the original Fern definition source files alongside the register call, we can follow-up with that change and plumb it through as another `Source`.

Reply:
That makes sense, would you be up to file a linear ticket for this on `Fern CLI` ?